openmc/plotter.py

```diff
@@ -0,0 +1,588 @@
+        Enrichment for U235 in weight percent. For example, input 4.95 for
+        4.95 weight percent enriched U. Default is None. This is only used for
+        items which are instances of openmc.Element
+    **kwargs
+        All keyword arguments are passed to
+        :func:`matplotlib.pyplot.figure`.
+
+    Returns
+    -------
+    fig : matplotlib.figure.Figure
+        If axis is None, then a Matplotlib Figure of the generated
+        cross section will be returned. Otherwise, a value of
+        None will be returned as the figure and axes have already been
+        generated.
+
+    """
+
+    from matplotlib import pyplot as plt
+
+    if isinstance(this, openmc.Nuclide):
+        data_type = 'nuclide'
+    elif isinstance(this, openmc.Element):
+        data_type = 'element'
+    elif isinstance(this, openmc.Material):
+        data_type = 'material'
+    else:
+        raise TypeError("Invalid type for plotting")
+
+    E, data = calculate_xs(this, types, temperature, sab_name, cross_sections,
+                           enrichment)
+
+    if divisor_types:
+        cv.check_length('divisor types', divisor_types, len(types),
+                        len(types))
+        Ediv, data_div = calculate_xs(this, divisor_types, temperature,
+                                      sab_name, cross_sections, enrichment)
+
+        # Create a new union grid, interpolate data and data_div on to that
+        # grid, and then do the actual division
+        Enum = E[:]
+        E = np.union1d(Enum, Ediv)
+        if data_type == 'nuclide':
+            data_new = []
+        else:
+            data_new = np.zeros((len(types), len(E)))
+
+        for line in range(len(types)):
+            if data_type == 'nuclide':
+                data_new.append(openmc.data.Combination([data[line],
+                                                         data_div[line]],
+                                                        [np.divide]))
+            else:
+                data_new[line, :] = \
+                    np.divide(np.interp(E, Enum, data[line, :]),
+                              np.interp(E, Ediv, data_div[line, :]))
+            if divisor_types[line] != 'unity':
+                types[line] = types[line] + ' / ' + divisor_types[line]
+        data = data_new
+
+    # Generate the plot
+    if axis is None:
+        fig = plt.figure(**kwargs)
+        ax = fig.add_subplot(111)
+    else:
+        fig = None
+        ax = axis
+    # Set to loglog or semilogx depending on if we are plotting a data
+    # type which we expect to vary linearly
+    if set(types).issubset(PLOT_TYPES_LINEAR):
+        plot_func = ax.semilogx
+    else:
+        plot_func = ax.loglog
+    # Plot the data
+    for i in range(len(data)):
+        if data_type == 'nuclide':
+            to_plot = data[i](E)
+        else:
+            to_plot = data[i, :]
+        if np.sum(to_plot) > 0.:
+            plot_func(E, to_plot, label=types[i])
```

Comment:
I was trying to plot `'nu-fission / absorption'` but the plot wasn't showing up because some of the values at the highest energies were NANs. I'm not sure what the best general solution is, but I found that adding `to_plot = np.nan_to_num(to_plot)` fixed it by replacing the NANs with 0.